### PR TITLE
Fix sending empty mac address to XO

### PR DESCRIPTION
**Pull request by ravager-dk**
The XO api no longer allows for Mac addresses to be empty when creating nics for a VM. However it does accept not specifying the MAC in which case it generates the MAC like before.

This changes introduces a check for mac address being empty and only appends it to the map if it contains a value

### CODE REVIEW

**Comment by ravager-dk** [comment]
> I suppose you saw that? [vatesfr/xen-orchestra#6758](https://github.com/vatesfr/xen-orchestra/pull/6758)

Yes, this does however not solve the issue for this provider. The change on xen-orchestra only changes how xo-web interacts with the API - empty mac address is removed in XO-web before communicating with the API. The same basically needs to be done for the terraform provider, which this PR solves.

File: xoa/resource_xenorchestra_vm.go [hunk 1]
```diff
@@ -230,15 +230,6 @@ func resourceVmSchema() map[string]*schema.Schema {
...
 						Computed: true,
-						StateFunc: func(val interface{}) string {
-							unformattedMac := val.(string)
-							mac, err := net.ParseMAC(unformattedMac)
-							if err != nil {
-								panic(fmt.Sprintf("Mac address `%s` was not parsable. This should never happened because Terraform's validation should happen before this is stored into state", unformattedMac))
-							}
-							return mac.String()
-
-						},
```
Comment (ddelnano) [inline xoa/resource_xenorchestra_vm.go:241]: @ravager-dk what was the rationale behind removing this? Doing so breaks the TestAccXenorchestraVm_createAndUpdateWithMacAddress [acceptance test](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/3e5215187f7ed75394b5d9fe6c5fbfb6f8646b65/xoa/resource_xenorchestra_vm_test.go#L674).

File: xoa/resource_xenorchestra_vm.go [hunk 2]
```diff
@@ -1184,20 +1183,19 @@ type guestNetwork map[string][]string
...
+//	    "ip":   []string{"interface 1's IPs",
+//	    "ipv4": []string{"interface 1's IPs",
+//	    "ipv6": []string{"ip1", "ip2"}
+//	  },
+//	  {
+//	    "ip":   []string{"interface 2's IPs",
+//	    "ipv4": []string{"interface 2's IPs",
+//	    "ipv6": []string{"ip1", "ip2"}
+//	  },
+//	}
```
Comment (ddelnano) [inline xoa/resource_xenorchestra_vm.go:1198]: Was this change intentional? Unless this is a gofmt or other formatting tool change, I think it would be better to leave this as is.

**Comment by ddelnano** [comment]
Thanks for contributing @ravager-dk! I have a few questions about the part of the change that is unrelated to the core issue you are trying to solve.

After those are discussed, I will run the acceptance test suite to make sure that everything is passing.

File: xoa/resource_xenorchestra_vm.go (hunk 1 above)
Comment (ravager-dk) [reply xoa/resource_xenorchestra_vm.go:241]: This got added into the PR by mistake from my other PR #233. Should I redo this PR completely without it, or can you edit this out?

File: xoa/resource_xenorchestra_vm.go (hunk 2 above)
Comment (ravager-dk) [reply xoa/resource_xenorchestra_vm.go:1198]: This is a formatting change by gofmt. Can be left out or stay in depending on your preference.

Comment (ddelnano) [reply xoa/resource_xenorchestra_vm.go:1198]: Sounds good. I'm fine with it applying then otherwise we will continue to undo it's changes.

File: xoa/resource_xenorchestra_vm.go (hunk 1 above)
Comment (ddelnano) [reply xoa/resource_xenorchestra_vm.go:241]: You don't need to redo the PR, but you do need to revert the change on the `empty-mac` branch and push to your remote again.

I can make the change if you want, but that would require opening a new PR (since I don't have commit access to your remote). I'm happy to do that if you prefer. If we go that route, we can make sure to preserve your commits so your contributions are still counted.

Comment (ravager-dk) [reply xoa/resource_xenorchestra_vm.go:241]: I will revert the the change and push it anew. :)

**Comment by ravager-dk** [comment]
> Thanks for contributing @ravager-dk! I have a few questions about the part of the change that is unrelated to the core issue you are trying to solve.
> 
> After those are discussed, I will run the acceptance test suite to make sure that everything is passing.

I've reverted the bad commit. Let me know if there are any further questions.

**Comment by ddelnano** [comment]
Thanks! No further questions. I'm running the acceptance tests now. Will let you know if there are any issues there.

**Comment by ddelnano** [comment]
The entire test suite passed so this is good to go. Thanks for fixing this!

**Comment by ddelnano** [comment]
This will be available in the v0.24.1 release. It should be available from the terraform registry in the next 30 mins.
